### PR TITLE
docs: update `CONTRIBUTING.md` guide

### PR DESCRIPTION
**Pull request by ttypic**
Added section about:

- building
- testing
- code style
- using local build as a dependency

## Summary by CodeRabbit

- **Documentation**
	- Expanded the `CONTRIBUTING.md` file with new sections on building the project, code standards, and running tests.
	- Added detailed instructions for using the Gradle wrapper, running Detekt for code style checks, and executing JUnit tests.
	- Enhanced clarity on using `chat-android` locally and updated the "Release Process" section for better understanding.

- **Chores**
	- Adjusted dependency resolution settings by removing `mavenLocal()` from the `build.gradle.kts` file.
	- Updated Detekt report configurations and ensured the `check` task runs the `detekt` task during checks.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -9,6 +9,100 @@
...
+
+```cmd
+gradlew.bat <task name>
+```
+
+To build the library’s AAR package, execute:
+
+```bash
+./gradlew chat-android:assemble
+```
```
Comment (sacOO7) [inline CONTRIBUTING.md:31]: The built AAR package will be available in the `chat-android/build/outputs/aar/` directory.

Comment (ttypic) [reply CONTRIBUTING.md:31]: fixed

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -9,6 +9,100 @@
 5. Push to the branch (`git push origin my-new-feature`)
 6. Create a new Pull Request
 
+### Building
+
+A gradle wrapper is included so these tasks can run without any prior installation of gradle. The Linux/OSX form of the commands, given
```
[the remark this answers is not in the document]
Comment (ttypic) [reply CONTRIBUTING.md]: fixed

**Review by sacOO7 — APPROVED** [review]
LGTM, I have picked up some suggestions from coderabbitai nitpick comments : P
You can check those comments again if needed. 
Thanks for adding CONTRIBUTING.md

**Pull request merged**